_resources/port1.0/group/python-1.0.tcl

```diff
@@ -447,3 +488,33 @@ pre-livecheck {
         livecheck.type  none
     }
 }
+
+# set-up a test phase
+pre-test {
+    if {[option python.test]} {
+        # return for py-<portname> stub ports, which should not have a test-phase enabled
+        if {[string match py-* [option name]] && ${name} eq ${subport}} {
+            return
+        }
+
+        test.run            yes
+        default test.target {}
+
+        if {[option python.test_framework] eq "pytest"} {
+            default test.args   "-o addopts=''"
+        }
+
+        # set PYTHONPATH depending on whether or not a port is "noarch"
+        if {[option supported_archs] eq "noarch"} {
+            default test.env    "PYTHONPATH=${worksrcpath}/build/lib"
+
+            # if using PEP517, unzip the .whl file to ${worksrcpath}/build/lib
+            if {${python.pep517}} {
+                file mkdir ${worksrcpath}/build/lib
+                system -W ${workpath} "unzip -o *.whl -d ${worksrcpath}/build/lib"
```

Comment:
Is this really the only way to make tests work? It seems completely against the spirit of PEP 517.

Reply:
I do agree that it's not really in the spirit of PEP 517 but I don't see a way around doing it like this. The `build` will generate the `.whl` file and we want to be able to run the test-phase before the actual `install` phase has happened. So the only thing we have at the beginning of the test-phase is this generated `.whl` file and in order to use the library it needs to be unpacked. Or am I missing something here / misunderstand what you mean?

Reply:
ah okay, I see your point. In many cases the files are indeed under `${worksrcpath}/build/lib*`. I did base it on one example I came across recently (`py-graphql-relay`) where there is no `build` directory generated, but only a `.whl` file. That specif port uses `build-backend = "poetry.core.masonry.api"` as backend - perhaps it's specific to that situation. Let me see if there are other ports that use the same backend and figure out if it only happens there - for the `setuptools` backend I realize now indeed that the files are always present in the canonical build directory.

Reply:
Poetry apparently expects tests to be run in a venv using `poetry run`. But it also only supports pure python projects currently, so it should always be possible to use the source directory in place.

Reply:
Yes, you're -of course- totally right about that. The issue I was having here is something specific to this port I think as it works fine with some other ports that use `poetry` as backend and where we indeed just use the "normal" `${worksrcpath}/build/lib`. So this was an unnecessary discussion I started... sorry about that!

Reply:
I wonder if the path(s) to use in `PYTHONPATH` need to be configurable in order to handle unusual cases like that. That can be future work, though.